add blog for integration with leapp (#151)
localstack/localstack.github.io: content/blog/2023-11-06-one-click-local-development-announcing-leapps-localstack-integration/index.md

---
title: "One Click Local Development: Announcing Leapp's LocalStack Integration"

> description: "We're excited to announce our partnership with Leapp by Noovolari to integrate LocalStack. Now you can easily create LocalStack sessions in Leapp to connect to the cloud emulator without hassles."

Reviewer: nit: I'd rather highlight the positive aspects ("seamless switch"), rather than emphasizing a negative one ("hassle") here.

Perhaps something like:
```suggestion
description: "We're excited to announce our partnership with Leapp by Noovolari to integrate LocalStack. Now you can easily create LocalStack sessions in Leapp and seamlessly switch between local dev and remote cloud environments."
```

lead: "We're excited to announce our partnership with Leapp by Noovolari to integrate LocalStack. Now you can easily create LocalStack sessions in Leapp to connect to the cloud emulator without hassles."
date: 2023-11-06
lastmod: 2023-11-06
images: []
contributors: ['Harsh Mishra']
leadimage: "localstack-leapp-cover-image.png"
tags: ['news']
show_cta_1: true
---

{{< img-simple src="localstack-leapp-cover-image.png" >}}

> [LocalStack](https://localstack.cloud) enables a fully functional cloud stack to emulate AWS cloud services on your local machine. It allows developers to shorten their feedback loop by enabling them develop, test, and deploy their applications locally while reducing costs and improving agility.  

Reviewer: ```suggestion
[LocalStack](https://localstack.cloud) enables a fully functional cloud stack to emulate AWS cloud services on your local machine. It allows developers to shorten their feedback loop by enabling them to develop, test, and deploy their applications locally while reducing costs and improving agility.  
```

LocalStack supports integrations, such as the [AWS CLI](https://docs.localstack.cloud/user-guide/integrations/aws-cli/), [CDK](https://docs.localstack.cloud/user-guide/integrations/aws-cdk/), [Terraform](https://docs.localstack.cloud/user-guide/integrations/terraform/), along with additional wrappers to make it possible for developers to connect to the local cloud emulator. However, maintaining these wrappers requires additional effort, and a lot of developers would like to switch from AWS to LocalStack in a click. 

> We’re excited to announce our official partnership and integration with [Leapp](https://leapp.cloud) by [Noovolari](https://www.noovolari.com/) to help you switch between AWS and LocalStack. You can use Leapp to create a LocalStack session on your developer machine that can then be used to set your local credential file and access your LocalStack resources, using standard integrations such as the AWS CLI.

Reviewer: ```suggestion
We’re excited to announce our partnership and integration with [Leapp](https://leapp.cloud) by [Noovolari](https://www.noovolari.com/) to help you switch between AWS and LocalStack. You can use Leapp to create a LocalStack session on your developer machine that can then be used to set your local credential file and access your LocalStack resources, using standard integrations such as the AWS CLI.
```

If not done already, you will be prompted to install the [AWS Session Manager Plugin](https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html). This plugin helps you to use the AWS CLI to start and end sessions to your managed instances.

### Setup the LocalStack session

You can now setup the LocalStack session. Click the **+** button on the top-bar to add a new session. Select **LocalStack** in the available options. A modal box will appear, and you will be prompted to add a **Session Alias**. Enter the desired alias and click on **Create Session**. Your LocalStack session is ready!

{{< img-simple src="localstack-session-leapp.png" >}}

### Run LocalStack with Leapp 

Your LocalStack session is ready! Select the session and click on **Start Session** in the bottom bar. You can notice that a **local** role has been attached to the session, along with a named profile and region.

{{< img-simple src="start-localstack-session.png" >}}

> Start your LocalStack container using your preferred method with `DEBUG=1` enabled. Navigate to your preferred terminal or Command Prompt and run the following set of commands:

Reviewer: ```suggestion
Start your LocalStack container using your preferred method with `DEBUG=1` enabled. Navigate to your preferred terminal or command prompt and run the following set of commands:
```
